### ISSUE TITLE
Spacing around T_CASE (creates fix loop)

### ISSUE THREAD

**Issue by spaceemotion** [slevomat/coding-standard#998]
With the default config on `SlevomatCodingStandard.ControlStructures.BlockControlStructureSpacing` I am encountering a problem with this snippet:

```php
switch ($foo) {
    case 'bar':
    case 'baz':
        return true;

    case 'lorem':
        return false;
}
```

It's going back and forth between the two messages depending on whether there's a space between the "bar" and "baz" cases:

```
Expected 1 lines after "case", found 0.
(SlevomatCodingStandard.ControlStructures.BlockControlStructureSpacing.IncorrectLinesCountAfterControlStructure)
```

```
Expected 0 lines before "case", found 1.
(SlevomatCodingStandard.ControlStructures.BlockControlStructureSpacing.IncorrectLinesCountBeforeFirstControlStructure)
```

I originally had the full PSR-12 ruleset enabled, but even with `PSR2.ControlStructures.SwitchDeclaration` excluded this still doesn't work and runs into a "fix-loop".

---

Edit: The only way I found to resolve this is by adding braces around the (empty) body, but this looks just (sorry) stupid:

```php
switch ($foo) {
    case 'bar': {
    }

    case 'baz':
        return true;

    case 'lorem':
        return false;
}
```

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/bfbf41690ade0f9cefd7e61df52ae9d87a52708c

**Comment by spaceemotion** [comment]
Hmm, will this add the space between them? Or does it also allow no space in-between?

**Comment by kukulich** [comment]
The default setting requires the space. However you can disable the sniff for `case` statements (enable only other control structures)

**Comment by spaceemotion** [comment]
I did this as a temporary fix for now, but I still want to keep the spaces after `break;`

**Comment by kukulich** [comment]
There's also `JumpStatementsSpacingSniff`

**Comment by spaceemotion** [comment]
Alright, thanks for the tip! I got the following config, but for some reason it now it's removing the spaces in between - instead of adding/leaving them be:

```xml
<rule ref="SlevomatCodingStandard.ControlStructures.JumpStatementsSpacing"/>

<rule ref="SlevomatCodingStandard.ControlStructures.BlockControlStructureSpacing">
  <!-- Remove case & default from the list -->
  <properties>
    <property name="tokensToCheck" type="array">
      <element value="T_IF"/>
      <element value="T_DO"/>
      <element value="T_WHILE"/>
      <element value="T_FOR"/>
      <element value="T_FOREACH"/>
      <element value="T_SWITCH"/>
      <element value="T_TRY"/>
    </property>
  </properties>
</rule>
```

creates:

```php
switch ($foo) {
    case 'bar':
    case 'baz':
        return true;
    case 'lorem':
        return false;
}
```
